.github/workflows/update_hub.yml

```diff
@@ -51,6 +51,15 @@ jobs:
           git add .
           git commit -m "Update MOC files" || echo "nothing to commit"
 
+      # This should be run after all the content additions have been made in case the structure of the vault changed.
+      - name: Update directory tree
+        run: |
+          cd .github/scripts
+          ./update_directory_tree.sh
+          cd ../..
+          git add .
+          git commit -m "Update directory tree" || echo "nothing to commit"
+
```

Comment:
I ran `./add_footer.py` on this branch, to see what would happen to the output, when viewed in Obsidian....

It inserted a "This note in GitHub block at the end of the tree structure, so the note then looked like this:

![image](https://user-images.githubusercontent.com/4840096/154918843-ed2bf4e8-165f-46b5-a4e5-e1f4e2709fad.png)

Elsewhere we have a workaround for this kind of thing with transcluded notes, by transcoding a specific heading rather than the whole file. I think that will look a bit ugly here, as we would have to add a heading to the top of the tree listing file.

Another alternative would be to edit `./add_footer.py` to skip the directory listing file...

Reply:
Or I could add a Heading and only transclude that part...? See my other comment. :)

Reply:
I'm unsure what the appearance would end up looking like if we added a heading to the transcluded file - so I can't really comment without seeing what it would look like in Obsidian preview...

Reply:
With an extra H1 (I just called it "Hub Tree") and the embed linking to `![[Hub Tree Structure#Hub Tree]]` it looks like this:

![220222_142252_screenshot](https://user-images.githubusercontent.com/69049131/155141168-e4fdbc06-20af-4cff-ac90-2ede1e177624.png)
![220222_142317_screenshot](https://user-images.githubusercontent.com/69049131/155141190-84c34767-6754-431a-aa24-3c00fc959cc6.png)

I'd actually prefer it that way, it's easy and looks better... Changing.

Reply:
Yes, it was the same way.

Reply:
Sorry, but I feel the level-1 "Hub Tree" heading looks out of place...

![image](https://user-images.githubusercontent.com/4840096/155198758-6fcaf66d-26c6-45c7-96f0-1af0db7b1b16.png)

Reply:
Before we go back and forth on this, please tell me if you want me to revert the commit or demote the heading. Thank you!

Reply:
> Before we go back and forth on this, please tell me if you want me to revert the commit or demote the heading. Thank you!

Great question....

I had a look at the heading levels, and I think a summary is:

- if you'd like there to be a heading at the top of the new file, then it probably needs to be level 4...
- otherwise, if you would like to exclude the file from add_footer.py, that would be fine too...

---

Here was my experiment... The heading levels on CONTRIBUTING.md are:

```
## Structure of the Community Vault

### Overview
This is a [tree](https://github.com/MrRaindrop/tree-cli) view of the basic folder structure of this Community Vault.

![[Hub Tree Structure#Hub Tree]]
```

So Overview is H3, and the tree seems to me to be logically inside that, which would make its heading $4.

Editing the transcluded file to this:

```#### Hub Tree
%% Hub Tree: Don’t edit below %%
....
```

Then gives this, which seems to flow OK:

![image](https://user-images.githubusercontent.com/4840096/155221180-4837f989-3297-47c1-9621-08578b96c92d.png)